core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -911,22 +928,29 @@ open class TileInfo {
             }
         }
 
+        val majorDepositFinal = majorDeposit ?: (rng.nextDouble() < approximateMajorDepositDistribution())
+        val depositAmounts = if (majorDepositFinal) newResource.majorDepositAmount else newResource.minorDepositAmount
         resourceAmount = when (tileMap.mapParameters.mapResources) {
-            MapResources.sparse -> {
-                if (majorDeposit) newResource.majorDepositAmount.sparse
-                else newResource.minorDepositAmount.sparse
-            }
-            MapResources.abundant -> {
-                if (majorDeposit) newResource.majorDepositAmount.abundant
-                else newResource.minorDepositAmount.abundant
-            }
-            else -> {
-                if (majorDeposit) newResource.majorDepositAmount.default
-                else newResource.minorDepositAmount.default
-            }
+            MapResources.sparse -> depositAmounts.sparse
+            MapResources.abundant -> depositAmounts.abundant
+            else -> depositAmounts.default
         }
     }
 
+    private fun approximateMajorDepositDistribution(): Double {
+        // We can't replicate the MapRegions resource distributor, so let's try to get
+        // a close probability of major deposits per tile
+        return getAllTerrains()
+            .flatMap {
+                it.getMatchingUniques(UniqueType.MajorStrategicFrequency)
+            }.sumOf {
+                // The unique param is literally "every N tiles", so to get a probability p=1/f
+                // The null hoops are so we can sum without exceptions no matter what a mod may put there 
+                it.params[0].toIntOrNull()?.run { if (this <= 0) null else 1.0 / this }
```

Comment:
Too clever by half, i'd much prefer it to be readable to be kotliners.

Reply:
Heh. 'kotliners' rhymes with 'flatliners'. First one was the better one. :+1: 
I thought hard on how to aggregate probabilities when you get frequencies, and gave up and left what worked. I'll reboot that thought.

Reply:
... much better. Though flatMap now exceeds polite line length a bit - marginally preferred this to flat~lining~mapping via double loop: less indent and the sequence initialization overhead is already there. The alternate would have been a val for the sequence, and that was 51:49 nay...